### PR TITLE
Improve Skaffold local dev process

### PR DESCRIPTION
**Pull request by shifucun**
- disable branch cache as the readiness probe will fail on Minikube (not sure why)
- make log working
- various fix for branch cache reload

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -41,10 +41,10 @@ which is compiled using [API Compiler](https://github.com/googleapis/api-compile
 ### Start mixer in minikube
 
 ```bash
-minikube start
+minikube start --meory 6G
```
Comment (pradh) [inline docs/developer_guide.md:44]: memory?

File: deploy/overlays/dev/patch_esp.yaml [hunk 2]
```diff
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+- op: add
+  path: /spec/template/spec/containers/0/args/0
```
Comment (pradh) [inline deploy/overlays/dev/patch_esp.yaml:16]: for my understanding, what are these indexes - 0/args/0, 1/args/0 ?

**Review by pradh — APPROVED** [review]

File: deploy/overlays/dev/patch_esp.yaml (hunk 2 above)
Comment (shifucun) [reply deploy/overlays/dev/patch_esp.yaml:16]: These corresponding to the deployment.yaml json tree. For example, here refers to the first container in deployment.yaml, and add an argument in the first position: https://github.com/datacommonsorg/mixer/blob/master/deploy/base/deployment.yaml#L53

File: docs/developer_guide.md (hunk 1 above)
Comment (shifucun) [reply docs/developer_guide.md]: good catch!

**Comment by shifucun** [comment]
thanks for review!

File: deploy/overlays/dev/patch_esp.yaml (hunk 2 above)
Comment (pradh) [reply deploy/overlays/dev/patch_esp.yaml:16]: Wow, seems brittle. Add a comment?

Comment (shifucun) [reply deploy/overlays/dev/patch_esp.yaml:17]: Done. And cleaned up the args

**Pull request merged**